src/Compilers/CSharp/Portable/Binder/Binder_Constraints.cs

```diff
@@ -34,7 +34,7 @@ internal partial class Binder
             // Create a map from type parameter name to ordinal.
             // No need to report duplicate names since duplicates
             // are reported when the type parameters are bound.
-            var names = new Dictionary<string, int>(n);
+            var names = new Dictionary<string, int>(n, StringOrdinalComparer.Instance);
```

Comment:
I assume that it only avoids doing an O(n) char-by-char string comparison in favor of an O(1) pointer comparison.